packages/web/src/common/store/premiumContent/sagas.ts

```diff
@@ -250,47 +255,64 @@ function* updateCollectibleGatedTrackAccess(
     )
   }
 
-  const trackMap = yield* call(getTokenIdMap, {
+  const { trackMap, skipped } = yield* call(getTokenIdMap, {
     tracks: allTracks,
     ethCollectibles: account.collectibleList || [],
     solCollectibles: account.solanaCollectibleList || [],
     premiumTrackSignatureIdSet
   })
+
+  // Set null for collectible gated track signatures as
+  // the user does not have nfts for those collections
+  // and therefore does not have access.
+  const premiumContentSignatureMap: {
+    [id: ID]: Nullable<PremiumContentSignature>
+  } = {}
+  Object.keys(allTracks).forEach((trackId) => {
+    const id = parseInt(trackId)
+    if (skipped.has(id)) return
+
+    const { premium_conditions: premiumConditions } = allTracks[trackId]
+    if (premiumConditions?.nft_collection && !trackMap[id]) {
+      premiumContentSignatureMap[id] = null
+    }
+  })
+  yield* put(updatePremiumContentSignatures(premiumContentSignatureMap))
+
   if (!Object.keys(trackMap).length) return
 
   // request premium content signatures for the relevant nft-gated tracks
   // which the client believes the user should have access to
   const apiClient = yield* getContext('apiClient')
 
-  const premiumContentSignatureMap = yield* call(
+  const premiumContentSignatureResponse = yield* call(
     [apiClient, apiClient.getPremiumContentSignatures],
     {
       userId: account.user_id,
       trackMap
     }
   )
 
-  // update premium content signatures
-  if (premiumContentSignatureMap) {
+  if (premiumContentSignatureResponse) {
+    const premiumContentSignatureMap: {
+      [id: ID]: Nullable<PremiumContentSignature>
+    } = { ...premiumContentSignatureResponse }
+    // Set null for tracks for which signatures did not get returned
+    // to signal that an attempt was made but the user does not have access.
+    Object.keys(trackMap).forEach((trackId) => {
+      const id = parseInt(trackId)
+      if (!premiumContentSignatureResponse[id]) {
+        premiumContentSignatureMap[id] = null
+      }
+    })
+
+    // update premium content signatures
     yield* put(updatePremiumContentSignatures(premiumContentSignatureMap))
 
-    // also update premium tracks' metadata with the newly obtained signature
-    yield* put(
-      cacheActions.update(
-        Kind.TRACKS,
-        Object.keys(premiumContentSignatureMap).map((trackId) => {
-          const id = parseInt(trackId)
-          return {
-            id,
-            metadata: {
-              // todo: remove below ts ignore
-              // @ts-ignore
-              premium_content_signature: premiumContentSignatureMap[id]
-            }
-          }
-        })
-      )
-    )
+    // we previously also updated the track's metadata in the store
```

Comment:
Just double-checking here that it's ok to not cache the signatures?

Reply:
yup thx